### DISCUSSION TITLE
Could `gix_object::Find` be async?

### DISCUSSION THREAD

**Discussion by NicholasLYang** [GitoxideLabs/gitoxide#1388] [Q&A] [answered]
### Summary 💡

Make the `gix_object::Find` trait require an async function. Or create an alternate async implementation for `LookupRefDeltaObjectsIter` that takes in an async version of `Find`.

### Motivation 🔦

I've been playing around with using gitoxide to build a git server backed by SQL. I'm resolving packfiles using `LookupRefDeltaObjectsIter`, but I need to run some async code in the `Find` trait, since I need to fetch from a database.

I'm down to contribute a PR if you're open to that!

**Comment by Byron** [comment, marked as answer]
For something to be `async`, it must satisfy certain conditions like:

* it does minimal work when polled
* it is cancellable

This is simply not true unless the operation is performed in a thread, something you can do with `tokio::spawn_blocking()` for instance.

Due to that, the non-transport parts won't ever be `async` - it would be a lie.
